### PR TITLE
Updates testsuite/MDAnalysisTests/analysis/test_nucleicacids.py

### PR DESCRIPTION
**Pull request by ooprathamm**
Changes made in this Pull Request:
Adds -
1.  Check wc_dist_shape
2.  Check if pair_distances is in wc_dist_results_keys
3.  Tests if ValueError is thrown if the residues given are not amino acids
4.  Tests if ValueError is thrown If the selections given are not the same length

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

**Comment by ooprathamm** [comment]
@orbeckst Please have a look at this by sparing some from your precious time :)
Let me know if you any questions.

File: testsuite/MDAnalysisTests/analysis/test_nucleicacids.py [hunk 1]
```diff
@@ -44,11 +44,37 @@ def wc_rna(u):
...
+    assert "pair_distances" in wc_rna.results.keys()
+
+
 def test_wc_dist(wc_rna):
     assert_allclose(wc_rna.results.pair_distances[0, 0], 4.3874702, atol=1e-3)
     assert_allclose(wc_rna.results.pair_distances[0, 1], 4.1716404, atol=1e-3)
 
 
+def test_wc_dist_invalid_residue_types(u):
+    with pytest.raises(ValueError, match=r"are not valid nucleic acids"):
```
Comment (richardjgowers) [inline testsuite/MDAnalysisTests/analysis/test_nucleicacids.py]: When writing tests to catch exceptions, you as few lines as possible inside the "with" block, ideally 1. This makes it more precise in what it catches and clearer what is getting tested

File: testsuite/MDAnalysisTests/analysis/test_nucleicacids.py [hunk 2]
```diff
@@ -44,11 +44,37 @@ def wc_rna(u):
     return WC
 
 
+def test_wc_dist_shape(wc_rna):
+    assert wc_rna.results.pair_distances.shape == (1, 2)
+
+
+def test_wc_dist_results_keys(wc_rna):
+    assert "pair_distances" in wc_rna.results.keys()
```
Comment (richardjgowers) [inline testsuite/MDAnalysisTests/analysis/test_nucleicacids.py]: Don't need the .keys for a dict "in" check

**Review by richardjgowers — CHANGES_REQUESTED** [review]
This looks good, couple fixes and we're good to go.

File: testsuite/MDAnalysisTests/analysis/test_nucleicacids.py (hunk 1 above)
Comment (RMeli) [inline testsuite/MDAnalysisTests/analysis/test_nucleicacids.py]:
```suggestion
    with pytest.raises(ValueError, match="are not valid nucleic acids"):
```

**Comment by ooprathamm** [comment]
As guided to have a single line in 'with' block.
test_wc_dist_invalid_residue_types throws flake error "local variable is assigned to but never used."

solution to this -
1. Add inline comment to ignore this error (done in the latest commit")
2. Add unnecessary .run()

@richardjgowers please guide which is a better solution.

**Comment by ooprathamm** [comment]
@richardjgowers Pardon couldn't see the simple approach.

**Comment by ooprathamm** [comment]
@richardjgowers Could you please look at the latest changes I made by sparing some from your important time. :)

**Review by richardjgowers — APPROVED** [review]

**Pull request merged**